cmd/fpga_crihook/main_test.go

```diff
@@ -0,0 +1,350 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+
+	"k8s.io/utils/exec"
+	fakeexec "k8s.io/utils/exec/testing"
+)
+
+func TestGetFPGAParams(t *testing.T) {
+	tcases := []struct {
+		stdinJSON      string
+		configJSON     string
+		expectedErr    bool
+		expectedRegion string
+		expectedAFU    string
+	}{
+		{
+			"stdin-correct.json",
+			"config-correct.json",
+			false,
+			"ce48969398f05f33946d560708be108a",
+			"f7df405cbd7acf7222f144b0b93acd18",
+		},
+		{
+			"stdin-broken-json.json",
+			"config-correct.json",
+			true,
+			"",
+			"",
+		},
+		{
+			"stdin-no-bundle.json",
+			"config-correct.json",
+			true,
+			"",
+			"",
+		},
+		{
+			"stdin-bundle-dir-doesnt-exist.json",
+			"config-correct.json",
+			true,
+			"",
+			"",
+		},
+		{
+			"stdin-correct.json",
+			"config-broken-json.json",
+			true,
+			"",
+			"",
+		},
+		{
+			"stdin-correct.json",
+			"config-no-process.json",
+			true,
+			"",
+			"",
+		},
+		{
+			"stdin-correct.json",
+			"config-no-env.json",
+			true,
+			"",
+			"",
+		},
+		{
+			"stdin-correct.json",
+			"config-no-region.json",
+			true,
+			"",
+			"",
+		},
+		{
+			"stdin-correct.json",
+			"config-no-afu.json",
+			true,
+			"ce48969398f05f33946d560708be108a",
+			"",
+		},
+	}
+	for _, tc := range tcases {
+		stdin, err := os.Open(path.Join("testdata", tc.stdinJSON))
+		if err != nil {
+			t.Fatalf("can't open file %s: %v", tc.stdinJSON, err)
+		}
+
+		he := newHookEnv("", tc.configJSON, nil)
+
+		region, afu, err := he.getFPGAParams(stdin)
+
+		if err != nil && !tc.expectedErr {
+			t.Errorf("unexpected error: %v", err)
+		} else if region != tc.expectedRegion {
+			t.Errorf("expected region: %s, actual: %s", tc.expectedRegion, region)
+		} else if afu != tc.expectedAFU {
+			t.Errorf("expected AFU: %s, actual: %s", tc.expectedAFU, afu)
+		}
+	}
+}
+
+func genFakeActions(fcmd *fakeexec.FakeCmd, num int) []fakeexec.FakeCommandAction {
+	var actions []fakeexec.FakeCommandAction
+	for i := 0; i < num; i++ {
+		actions = append(actions, func(cmd string, args ...string) exec.Cmd {
+			return fakeexec.InitFakeCmd(fcmd, cmd, args...)
+		})
+	}
+	return actions
+}
+
+func TestValidateBitstream(t *testing.T) {
+	tcases := []struct {
+		fpgaRegion  string
+		fpgaAfu     string
+		expectedErr bool
+		fakeAction  []fakeexec.FakeCombinedOutputAction
+	}{
+		{
+			"ce48969398f05f33946d560708be108a",
+			"f7df405cbd7acf7222f144b0b93acd18",
+			false,
+			[]fakeexec.FakeCombinedOutputAction{
+				func() ([]byte, error) {
+					return ioutil.ReadFile("testdata/gbs-info-correct.json")
+				},
+			},
+		},
+		{
+			"",
+			"",
```

Comment:
If a struct has more than 2 fields it makes sense to initialize its values with field names like
```
{
    expectedErr: true,
    fakeAction: []fakeexec.FakeCombinedOutputAction{...},
},
```
then 1) empty value can be omitted and 2) it's easier to read the code when the struct's definition is out of editor's frame.